README.md

```diff
@@ -114,7 +115,25 @@ this.setState({ refreshing: true }, () => {
 });
 ```
 
-## Best Practices
+## Error handling
+
+Error handling is complicated in general - especially in GraphQL environment. There are several scenarios that may occur:
+
+1. GraphQL API returns `data` field and no `errors`
+
+This should be considered as a valid full response and there should not be any errors. There may be nullable fields, however.
+
+2. GraphQL API returns `data = null` and `errors` field
```

Comment:
I'm trying to test this scenario by modifying the fetchQuery:

```
  const fetchQuery = async (operation, variables, cacheConfig) => {
    return {
      data: null,
      errors: [
        {
          message: 'Some message',
        },
      ],
    };
```

Then I get the following message, is this the expected behavior? Or am I testing it wrong (and how can I test it correctly)?

![simulator screen shot - iphone 5s - 2017-12-28 at 15 21 40](https://user-images.githubusercontent.com/1473642/34413307-d2f31efe-ebe2-11e7-9371-d98119188453.png)

Reply:
Yes, this is correct behavior. Data cannot be `null` because this indicates a fatal GraphQL error and not partial error. See documentation. This is the only situation when Relay actually works correctly and handles errors...

Reply:
Ah, you are pointing to the docs. So yes, this is `GeneralError` component in action.